### PR TITLE
feat(integrations): Add integration with MonAI

### PR DESCRIPTION
**Pull request by soumik12345**
Description
-----------
Add integration with MonAI PyTorch Ignite Handlers.
- [x] Add `WandbStatsHandler` for MonAI Ignite trainer, for logging metrics during training and validation.
- [x] Add `WandbModelCheckpoint` handler for MonAI Ignite trainer, for logging model checkpoints during training.

Usage Example for `WandbStatsHandler`
---

```python
# For training
train_wandb_stats_handler = WandbStatsHandler(output_transform=lambda x: x)
train_wandb_stats_handler.attach(trainer)
# For evaluation
val_wandb_stats_handler = WandbStatsHandler(
    output_transform=lambda x: None,
    global_epoch_transform=lambda x: trainer.state.epoch,
)
val_wandb_stats_handler.attach(evaluator)
```

Usage Example for `WandbModelCheckpoint`
---

```python
checkpoint_handler = WandbModelCheckpoint(
    "./runs_dict/", "net", n_saved=10, require_empty=False
)
trainer.add_event_handler(
    event_name=Events.EPOCH_COMPLETED,
    handler=checkpoint_handler,
    to_save={"net": net, "opt": opt},
)
```
 
Testing
-------
How was this PR tested?

### CODE REVIEW

**Review by dmitryduev — CHANGES_REQUESTED** [review]
Looks great! you need to add a new shard for the functional tests to run (see the missing coverage). Also, the yea config is invalid, which trips over all the func tests.

File: wandb/integration/monai/stats_handler.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+                which logs the iteration number and the output value in `engine.state.output` (if any).
+            output_transform: (Callable) a callable that is used to transform the ignite.engine.state.output into the value to log.
+                For example, if `ignite.engine.state.output` is a tuple `(loss, y_pred, y)`, then use
+                `output_transform=lambda x: x[0]` to log `loss`, or use `output_transform=lambda x: x[1:]` to log `(y_pred, y)`.
+                Default is `lambda x: x[0]` which corresponds to logging `loss`.
+            global_epoch_transform: (Callable) a callable that is used to transform the ignite.engine.state.epoch into the value
+                to log. This can be used to log the global step number across multiple epochs. Default is `lambda x: x`.
+            state_attributes: (Union[Sequence[str], None]) a list of attributes of `ignite.engine.state` that will be logged.
+                If None, nothing will be logged. Default is None.
+            tag_name: (str) the name of the loss tag to log. Default is "Loss".
```
Comment (ayulockin) [inline wandb/integration/monai/stats_handler.py:74]: > the name of the loss tag to log

a bit unclear

File: wandb/integration/monai/stats_handler.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            bool, Callable[[ignite_engine.Engine, int], bool], int
+        ] = True,
+        epoch_log: Union[bool, Callable[[ignite_engine.Engine, int], bool], int] = True,
+        epoch_event_logger: Union[
+            Callable[[ignite_engine.Engine, Any], Any], None
+        ] = None,
+        iteration_event_logger: Union[
+            Callable[[ignite_engine.Engine, Any], Any], None
+        ] = None,
+        output_transform: Callable = lambda x: x[0],
```
Comment (ayulockin) [inline wandb/integration/monai/stats_handler.py:26]: I think this is an important argument and many a times the default setting might not be sufficient. I think this should show up earlier.

File: tests/functional_tests/t0_main/monai/test_monai_classification.yea [hunk 3]
```diff
@@ -0,0 +1,32 @@
+id: 0.monai.classification
+plugin:
+    - wandb
+tag:
+  shard: standalone-gpu
```
Comment (dmitryduev) [inline tests/functional_tests/t0_main/monai/test_monai_classification.yea:5]:
```suggestion
```

**Review by dmitryduev — CHANGES_REQUESTED** [review]
Thanks @soumik12345. I ran the tests on the default shard and it's erroring: https://app.circleci.com/pipelines/github/wandb/wandb/30457/workflows/fdefca75-d2e1-421f-8466-ba0625d42e0b/jobs/986798?invite=true#step-105-680
can you please look into that?

**Review by dmitryduev — CHANGES_REQUESTED** [review]
Looks like the relevant test is failing: https://app.circleci.com/pipelines/github/wandb/wandb/30522/workflows/84591fb4-d52e-4256-9ae6-3d2515a885e8/jobs/990365?invite=true#step-105-381
So, moving this PR to the next milestone.

**Review by dmitryduev — CHANGES_REQUESTED** [review]
Looks like the relevant test is failing: https://app.circleci.com/pipelines/github/wandb/wandb/30522/workflows/84591fb4-d52e-4256-9ae6-3d2515a885e8/jobs/990365?invite=true#step-105-381
So, moving this PR to the next milestone.

File: wandb/integration/monai/model_checkpoint.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        kwargs: (Any) Accepted keyword arguments for `torch.save` or `xm.save` in `DiskSaver`.
+    """
+
+    def __init__(
+        self,
+        dirname: Union[str, Path],
+        filename_prefix: str = "",
+        score_function: Optional[Callable] = None,
+        score_name: Optional[str] = None,
+        n_saved: Union[int, None] = 1,
```
Comment (kptkin) [inline wandb/integration/monai/model_checkpoint.py:64]:
```suggestion
        n_saved: Optional[int] = 1,
```

File: wandb/integration/monai/model_checkpoint.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
+from pathlib import Path
+from typing import Any, Callable, Mapping, Optional, Union
+
+from ignite.engine import Engine
+
+import wandb
+from wandb.util import get_module
+
+ignite_engine = get_module("ignite.engine")
+ignite_handlers = get_module("ignite.handlers")
```
Comment (kptkin) [inline wandb/integration/monai/model_checkpoint.py:10]: what happens if this is not available, i think this code will fail. Try running this without ignite installed. 
Also why do you have on line 4 direct import and on line 9 you use get_module?

File: tests/functional_tests/t0_main/monai/test_monai.yea [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    - :op:==:
+        - :wandb:runs[0][summary][epoch/epoch]
+        - 2
+    - :op:>=:
+        - :wandb:runs[0][summary][iteration/Loss]
+        - 0
+    - :op:==:
+        - :wandb:runs[0][summary][iteration/iteration]
+        - 20
+    - :wandb:runs[0][exitcode]: 0
```
Comment (dmitryduev) [inline tests/functional_tests/t0_main/monai/test_monai.yea:31]: Curiously, it looks like it fails on ci, although yea doesn't seem to catch it:
https://app.circleci.com/pipelines/github/wandb/wandb/35609/workflows/51972e1e-d502-4eda-893d-e05d98e6dfb6/jobs/1201297?invite=true#step-105-13848_46

**Review by dmitryduev — CHANGES_REQUESTED** [review]

**Comment by dmitryduev** [comment]
Closing as stale and likely moot.
